lib/utils.js

```diff
@@ -132,7 +132,9 @@ exports.compact = function (obj, references) {
         var compacted = [];
 
         for (var i = 0; i < obj.length; ++i) {
-            if (typeof obj[i] !== 'undefined') {
+            if (Object.prototype.toString.call(obj[i]) === '[object Object]') {
```

Comment:
Check for arrays is fine but check for functions is in my opinion not nescessary, because you can't define functions in query string.